### PR TITLE
A simple demo of how the most rudimentary form of blockchain works.

### PR DESCRIPTION
**Pull request by anuran-roy**
#### Description of Change

I have implemented a simple demo of the most basic form of a blockchain using C++. It involves a simple PoW algorithm, a User Class, a Ledger class, a Node class, and a Transaction class.

#### Checklist

- [X] Added description of change
- [X] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [X] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [X] Relevant documentation/comments is changed or added
- [X] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [X] Search previous suggestions before making a new one, as yours may be a duplicate.
- [X] I acknowledge that all my contributions will be made under the project's license.

Notes: None whatsoever

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1644"></a>

### CODE REVIEW

**Comment by anuran-roy** [comment]
Uhh... @Panquesito7  can you please assign someone to review this PR?

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thank you for submitting a PR! 😄 Sadly, this does not meet our requirements. Check out our [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Please read them **carefully** and **follow them**. **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂

File: search/text_search.cpp [hunk 1]
```diff
@@ -12,31 +12,65 @@
 
 /** Main function
  */
+
+std::string lower(std::string word) {
+    int length = word.length();
```
Comment (Panquesito7) [inline search/text_search.cpp:17]: Please rever the changes in this file or make a separate PR for this.

File: blockchain/blockchain.cpp [hunk 2]
```diff
@@ -0,0 +1,349 @@
+/*
```
Comment (Panquesito7) [inline blockchain/blockchain.cpp:1]: Your code is missing proper documentation (see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program)).

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: search/text_search.cpp (hunk 1 above)
Comment (anuran-roy) [reply search/text_search.cpp:17]: I'm really sorry for this mistake. I'll make a new PR right away 😃
